core/src/com/biglybt/core/networkmanager/impl/tcp/TCPTransportHelper.java

```diff
@@ -285,43 +274,7 @@ public TCPTransportHelper( SocketChannel _channel ) {
 
 			// log( buffers, array_offset, length );
 
-			if ( enable_efficient_io && remainingBytesToScatter < 1 ){
-
-				try{
-					written_sofar = channel.write( buffers, array_offset, length );
-
-				}catch( IOException ioe ) {
-
-					//a bug only fixed in Tiger (1.5 series):
-					//http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=4854354
-					String msg = ioe.getMessage();
-					if( msg != null && msg.equals( "A non-blocking socket operation could not be completed immediately" ) ) {
-						enable_efficient_io = false;
-						Logger.log(new LogAlert(LogAlert.UNREPEATABLE, LogAlert.AT_WARNING,
-								"WARNING: Multi-buffer socket write failed; "
-								+ "switching to single-buffer mode.\n"
-								+ "Upgrade to JRE 1.5 (5.0) series to fix this problem!"));
-					}
-					throw ioe;
-				}
-			}else{
-
-					//single-buffer mode
-
-				for( int i=array_offset; i < (array_offset + length); i++ ) {
-
-					int data_length = buffers[ i ].remaining();
-
-					int written = channelWrite(buffers[i]);
```

Comment:
`channelWrite()` is the only place that modifies `remainingBytesToScatter`. When using efficient_io mode then this was left unaltered by this class. 
Only other entry point is thru `public void setScatteringMode(long forBytes)`.

My first imression is that if you're in `enable_efficiant_io` mode then you're bound to it.